content/platforms/kubernetes/tasks/using-kustomnize.md

```diff
@@ -0,0 +1,167 @@
+
+For example, production clusters will likely require more resources and greater limits than development or testing systems. In addition, your secrets, passwords, and other security settings are likely to differ across deployment environments. Kustomize makes it easier to handle these differences between environments.
+
+## A simple example
+
+Kustomize works on the building variants from a hierarchy of configurations. In the simplest setup, there is one “base” directory containing all of the common resources. For each variant, there are different directories, each with instructions on how to produce the variant.
+
+For example, say we want to control the limits and requests for a node so that they are different values for development and production. To start, we must create a “base” directory that contains a default configuration for a simple cluster:
+
+```
+mkdir -p base
+cat << EOF > base/cluster.yaml
+apiVersion: app.redislabs.com/v1
+kind: RedisEnterpriseCluster
+metadata:
+  name: test
+spec:
+  nodes: 3
+  username: "app@example.org"
+EOF
+```
+
+We then need to configure the “base” directory to be recognized by kustomize. The simplest configuration lists only the resources:
+
+```
+cat << EOF > base/kustomization.yaml
+resources:
+ - cluster.yaml
+EOF
+```
+
+Now we can create a variant for development that tunes down the CPU requirements and also creates a single-node cluster while changing the namespace:
+
+```
+mkdir -p dev
+cat << EOF > dev/cluster.yaml
+apiVersion: app.redislabs.com/v1
+kind: RedisEnterpriseCluster
+metadata:
+  name: test
+spec:
+  nodes: 1
+  redisEnterpriseNodeResources:
+    limits:
+      cpu: 1
+      memory: 4Gi
+    requests:
+      cpu: 1
+      memory: 4Gi
+EOF
+```
+
+In this example, the cluster resource must repeat the prolog metadata so that kustomize can match our resources. At minimum, we need `apiVersion`, `kind`, and `metadata.name`.
+
+As before, we need to tell kustomize about our resources using a `kustomization.yaml` file. In this case, we need to tell kustomize what to do with our variant of `cluster.yaml`. Specifically, we want to patch the base `cluster.yaml` by strategically merging properties:
+
+```
+cat << EOF > dev/kustomization.yaml
+bases:
+ - ../base
+
+namespace: dev
+
+patchesStrategicMerge:
+ - cluster.yaml
+EOF
+```
+
+For a QA deployment, we might need more resources:
+
+```
+mkdir -p qa
+cat << EOF > qa/cluster.yaml
+apiVersion: app.redislabs.com/v1
+kind: RedisEnterpriseCluster
+metadata:
+  name: test
+spec:
+  nodes: 5
+  redisEnterpriseNodeResources:
+    limits:
+      cpu: 2
+      memory: 25Gi
+    requests:
+      cpu: 2
+      memory: 25Gi
+EOF
+cat << EOF > qa/kustomization.yaml
+bases:
+ - ../base
+
+namespace: qa
+
+patchesStrategicMerge:
+ - cluster.yaml
+EOF
+```
+
+At this, we can test our “dev” customization by running the folliwing:
+
+```
+kubectl kustomize dev
```

Comment:
This confused me, since you earlier say to run it using `kubectl apply -k`

Reply:
`kubectl apply -k` applys the configuration to the cluster and `kubectl kustomize dev` just runs the kustomize process locally and shows you the output.

Reply:
I fixed that sentence now.